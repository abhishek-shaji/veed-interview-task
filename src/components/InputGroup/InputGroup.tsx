import React from 'react';

interface InputGroupPropType {
  label?: string;
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  placeholder: string;
  value?: string;
}

const InputGroup = ({
  label,
  placeholder,
  onChange,
  value,
}: InputGroupPropType) => (
  <div className="w-full">
    {label && (
      <label
        htmlFor="email"
        className="block mb-1.5 ml-1 text-lg font-semibold w-full"
        data-testid="input-group-label"
      >
        {label}
      </label>
    )}
    <input
      type="text"
      className="bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-xl focus:ring-blue-500 focus:border-blue-500 block w-full p-3"
      placeholder={placeholder}
      required
      onChange={onChange}
      value={value}
    />
  </div>
);

export { InputGroup };
