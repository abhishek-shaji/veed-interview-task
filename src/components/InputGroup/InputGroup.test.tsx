// write tests using jest for RepositoryList component
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { InputGroup } from '@/components/InputGroup/InputGroup';

describe('InputGroup', () => {
  it('should render correctly', () => {
    const { container } = render(
      <InputGroup
        onChange={() => {}}
        placeholder="test placeholder"
        value={''}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render correctly with value', () => {
    const { container } = render(
      <InputGroup
        onChange={() => {}}
        placeholder="test placeholder"
        value="test value"
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render with label', () => {
    const { container } = render(
      <InputGroup
        label="test label"
        onChange={() => {}}
        placeholder="test placeholder"
        value="test value"
      />,
    );

    const element = screen.getByTestId('input-group-label');

    expect(container).toContainElement(element);

    expect(container).toMatchSnapshot();
  });
});
