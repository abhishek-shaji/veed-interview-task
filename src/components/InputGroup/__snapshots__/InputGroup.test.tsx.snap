// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputGroup should render correctly 1`] = `
<div>
  <div
    class="w-full"
  >
    <input
      class="bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-xl focus:ring-blue-500 focus:border-blue-500 block w-full p-3"
      placeholder="test placeholder"
      required=""
      type="text"
      value=""
    />
  </div>
</div>
`;

exports[`InputGroup should render correctly with value 1`] = `
<div>
  <div
    class="w-full"
  >
    <input
      class="bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-xl focus:ring-blue-500 focus:border-blue-500 block w-full p-3"
      placeholder="test placeholder"
      required=""
      type="text"
      value="test value"
    />
  </div>
</div>
`;

exports[`InputGroup should render with label 1`] = `
<div>
  <div
    class="w-full"
  >
    <label
      class="block mb-1.5 ml-1 text-lg font-semibold w-full"
      data-testid="input-group-label"
      for="email"
    >
      test label
    </label>
    <input
      class="bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-xl focus:ring-blue-500 focus:border-blue-500 block w-full p-3"
      placeholder="test placeholder"
      required=""
      type="text"
      value="test value"
    />
  </div>
</div>
`;
